pkg/network/tracer/cilium_lb.go

```diff
@@ -0,0 +1,270 @@
+}
+
+func ntohs(n uint16) uint16 {
+	return binary.BigEndian.Uint16([]byte{byte(n), byte(n >> 8)})
+}
+
+func htons(n uint16) uint16 {
+	b := make([]byte, 2)
+	binary.BigEndian.PutUint16(b, n)
+	return *(*uint16)(unsafe.Pointer(&b[0]))
+}
+
+func (clb *ciliumLoadBalancerConntracker) updateBackends() {
+	clb.m.Lock()
+	defer clb.m.Unlock()
+
+	it := clb.backends.Iterate()
+	var k backend4KeyV3
+	var v backend4ValueV3
+	for it.Next(&k, &v) {
+		clb.backendIDToBackend[k.ID] = backend{
+			addr: util.AddressFromNetIP(net.IPv4(v.Address[0], v.Address[1], v.Address[2], v.Address[3])),
+			port: ntohs(v.Port),
+		}
+	}
+
+	if it.Err() != nil {
+		log.Warnf("error iterating lb backends map: %s", it.Err())
+	}
+}
+
+func (clb *ciliumLoadBalancerConntracker) Describe(chan<- *prometheus.Desc) {
+}
+
+// Collect returns the current state of all metrics of the collector
+func (clb *ciliumLoadBalancerConntracker) Collect(chan<- prometheus.Metric) {
+}
+
+func (clb *ciliumLoadBalancerConntracker) GetTranslationForConn(c *network.ConnectionTuple) *network.IPTranslation {
+	// TODO: add ipv6 support
+	if c.Family != network.AFINET {
+		return nil
+	}
+
+	if c.Direction != network.OUTGOING {
+		return nil
+	}
+
+	if c.Dest.IsLoopback() {
+		return nil
+	}
+
+	queryMap := clb.ctTCP
+	t := tupleKey4{
+		Flags:      TUPLE_F_OUT | TUPLE_F_SERVICE,
+		NextHeader: uint8(unix.IPPROTO_TCP),
+		SourcePort: htons(c.SPort),
+		DestPort:   htons(c.DPort),
+		SourceAddr: c.Source.As4(),
+		DestAddr:   c.Dest.As4(),
+	}
+	if c.Type == network.UDP {
+		t.NextHeader = unix.IPPROTO_UDP
+		queryMap = clb.ctUDP
+	}
+
+	log.TraceFunc(func() string {
+		return fmt.Sprintf("looking up tuple %+v in ct map", t)
+	})
+
+	var ctEntry ctEntry
+	var err error
+	if err = queryMap.Lookup(&t, &ctEntry); err != nil {
+		if !errors.Is(err, ebpf.ErrKeyNotExist) {
+			log.Warnf("error looking up %+v in ct map: %s", t, err)
+		}
+
+		log.TraceFunc(func() string {
+			return fmt.Sprintf("lookup failed for %+v in ct map", t)
+		})
+
+		return nil
+	}
+
+	log.TraceFunc(func() string {
+		return fmt.Sprintf("found ct entry for %+v: %+v", t, ctEntry)
+	})
+
+	clb.m.Lock()
+	defer clb.m.Unlock()
+
+	if b, ok := clb.backendIDToBackend[uint32(ctEntry.BackendID)]; ok && (b.addr != c.Dest || b.port != c.DPort) {
+		return &network.IPTranslation{
+			ReplDstIP:   c.Source,
+			ReplDstPort: c.SPort,
+			ReplSrcIP:   b.addr,
+			ReplSrcPort: b.port,
+		}
+	}
+
+	return nil
+}
+
+// GetType returns a string describing whether the conntracker is "ebpf" or "netlink"
+func (clb *ciliumLoadBalancerConntracker) GetType() string {
+	return "cilium_lb"
+}
+
+func (clb *ciliumLoadBalancerConntracker) DeleteTranslation(*network.ConnectionTuple) {
+}
+
+func (clb *ciliumLoadBalancerConntracker) DumpCachedTable(context.Context) (map[uint32][]netlink.DebugConntrackEntry, error) {
+	return nil, nil
+}
+
+func (clb *ciliumLoadBalancerConntracker) Close() {
+	clb.stop <- struct{}{}
+	<-clb.stop
+	clb.ctTCP.Map().Close()
```

Comment:
I believe in other places in the code we wrap certain cleanup in a sync.Once so that in case close is called multiple times, we can avoid panics. Would it be useful to do that here?